File: app/src/main/java/jensklingenberg/de/tuxoid/model/Element/Destination/Background.java
package jensklingenberg.de.tuxoid.model.Element.Destination;

import android.graphics.Bitmap;

import jensklingenberg.de.tuxoid.model.MyImage;

/**
 * Created by jens on 18.04.17.
 */

public class Background extends Destination {


    private Bitmap image;
    private int type;

    public Background(int type, int z, int y, int x) {
        super(type, z, y, x);
        this.type = type;
        this.image = MyImage.getImage(type);
    }


    @Override
    public Bitmap getImage() {
        return image;
    }

    @Override
    public int getType() {
        return type;
    }
}

File: app/src/main/java/de/jensklingenberg/tuxoid/model/element/Background.kt
package de.jensklingenberg.tuxoid.model.element

import de.jensklingenberg.tuxoid.R
import de.jensklingenberg.tuxoid.interfaces.IReachable

class Background : Element(ElementType.BACKGROUND){
    override val imageResId: Int = R.drawable.background
    override fun isReachable()=true
}